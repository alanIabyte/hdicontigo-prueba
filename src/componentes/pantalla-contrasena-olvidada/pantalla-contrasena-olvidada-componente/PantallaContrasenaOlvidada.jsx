import React, { useState, useEffect } from "react";
import { useHistory } from "react-router-dom";
import { v4 } from "uuid";
import { loader } from "graphql.macro";
import { useMutation } from "@apollo/react-hooks";
import {
  CuerpoContrasenaOlvidada,
  ContenedorBoton,
} from "./PantallaContrasenaOlvidada.styled";
import {
  EnvolvedorPantalla,
  Pantalla,
} from "../../componentes-styled-compartidos/Pantalla.styled";
import Enviado from "./PantallaContrasenaOlvidadaEnviado";
import IndicadorCarga from "../../indicador-carga";
import Encabezado from "../../encabezado";
import Boton from "../../boton";
import BarraAlerta from "../../barra-alerta";
import CampoTexto from "../../campo-texto";

const valores = {
  telefono: "",
};

const OLVIDE_CONTRASENA = loader(
  "../../../graphQL/mutation/seguridad/olvide_contrasena.graphql"
);

const PantallaContrasenaOlvidada = () => {
  const history = useHistory();
  const [botonDeshabilitado, asignarValorBotonDeshabilitado] = useState(true);
  const [mostrarBarraAlerta, asignarValorMostrarBarraAlerta] = useState(false);
  const [pantalla, asignarValorPantalla] = useState("");
  const [cargando, asignarValorCargando] = useState(false);

  const diccionario = {
    encabezado: "Olvidé mi contraseña",
    cuerpo:
      // eslint-disable-next-line max-len
      "Ingresa el teléfono vinculado a tu cuenta. Enviaremos un WhatsApp/SMS con el enlace para restablecer tu contraseña.",
    etiquetaTelefono: "Teléfono",
    etiquetaBoton: "Enviar enlace",
    mensajeAlerta:
      "Tenemos problemas para realizar el cambio de contraseña. Favor de intentarlo mas tarde.",
  };

  const cambiarPantalla = (pantallaACambiar) => {
    asignarValorPantalla(pantallaACambiar);
  };

  const [enviarContrasenaOlvidada, { data, loading }] =
    useMutation(OLVIDE_CONTRASENA);

  useEffect(() => {
    if (valores.telefono === "") {
      asignarValorBotonDeshabilitado(true);
    } else {
      asignarValorBotonDeshabilitado(false);
    }
  }, []);

  useEffect(() => {
    if (!loading && data) {
      asignarValorCargando(false);
      if (
        data &&
        data.olvide_contrasena &&
        data.olvide_contrasena.mensaje &&
        data.olvide_contrasena.mensaje === "IDX10023"
      ) {
        // es por que trono un servicio pintar alerta
        asignarValorMostrarBarraAlerta(true);
        asignarValorBotonDeshabilitado(true);
        setTimeout(() => {
          asignarValorBotonDeshabilitado(false);
        }, 4000);
      } else {
        cambiarPantalla("enviado");
      }
    } else if (loading) {
      asignarValorCargando(true);
    }
  }, [data, loading]);

  const mandarPeticion = () => {
    enviarContrasenaOlvidada({
      variables: {
        usuario: valores.telefono,
      },
    });
  };

  const alCambiarTelefono = (evento) => {
    if (evento) {
      const valor = evento.target.value;
      if (valor === "") {
        asignarValorBotonDeshabilitado(true);
      } else {
        asignarValorBotonDeshabilitado(false);
      }
      valores.telefono = valor;
    }
  };

  let pantallaAMostrar = (
    <EnvolvedorPantalla key={v4()} id="PantallaContrasenaOlvidada">
      {cargando ? <IndicadorCarga /> : null}
      <Encabezado
        titulo={diccionario.encabezado}
        funcionRegresar={() => {
          history.push("/");
        }}
      />
      <Pantalla>
        <BarraAlerta
          etiqueta={diccionario.mensajeAlerta}
          mostrarAlerta={mostrarBarraAlerta}
          manejarCierre={asignarValorMostrarBarraAlerta}
          estilo="error"
        />
        <CuerpoContrasenaOlvidada>
          {diccionario.cuerpo}
        </CuerpoContrasenaOlvidada>
        <CampoTexto
          id="campoTelefono"
          etiqueta={diccionario.etiquetaTelefono}
          valor={valores.telefono}
          numeroDeCaracteres={15}
          esNumerico
          enCambio={alCambiarTelefono}
          autoenfoque
        />
        <ContenedorBoton>
          <Boton
            etiqueta={diccionario.etiquetaBoton}
            tema="estandar"
            enClick={mandarPeticion}
            deshabilitado={botonDeshabilitado}
          />
        </ContenedorBoton>
      </Pantalla>
    </EnvolvedorPantalla>
  );

  if (pantalla === "enviado") {
    pantallaAMostrar = <Enviado telefono={valores.telefono} />;
  }
  return pantallaAMostrar;
};

export default PantallaContrasenaOlvidada;
