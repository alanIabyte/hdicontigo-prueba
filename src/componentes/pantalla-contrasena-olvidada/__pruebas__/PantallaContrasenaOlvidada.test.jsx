import React from "react";
import renderer from "react-test-renderer";
import PantallaContrasenaOlvidada from "../pantalla-contrasena-olvidada-componente/PantallaContrasenaOlvidada";

jest.mock("@apollo/react-hooks", () => ({
  useMutation: jest.fn().mockReturnValue([{}, { loading: false }]),
}));

describe("#PantallaContrasenaOlvidada", () => {
  it("se despliega correctamente", () => {
    const tree = renderer.create(<PantallaContrasenaOlvidada />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
