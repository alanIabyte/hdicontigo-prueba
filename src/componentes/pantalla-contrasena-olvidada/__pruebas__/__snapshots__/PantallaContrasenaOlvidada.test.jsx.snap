// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#PantallaContrasenaOlvidada se despliega correctamente 1`] = `
<div
  className="sc-jsJBEP hrEzkG"
  id="PantallaContrasenaOlvidada"
>
  <div>
    Soy un contenedor simulado
  </div>
  <div
    className="sc-koXPp gBlQG"
  >
    <div
      className="sc-kOHTFB hniLxw"
    >
      <div
        className="sc-dtInlm QafKl"
      >
        <div
          className="sc-kOPcWz juGHSd"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 11c-.55 0-1-.45-1-1V8c0-.55.45-1 1-1s1 .45 1 1v4c0 .55-.45 1-1 1zm1 4h-2v-2h2v2z"
            />
          </svg>
        </div>
        <span
          className="sc-cWSHoV ckYGBL"
          dangerouslySetInnerHTML={
            Object {
              "__html": "Tenemos problemas para realizar el cambio de contraseña. Favor de intentarlo mas tarde.",
            }
          }
        />
        <button
          className="sc-eBMEME ea-dXxx"
          onClick={[Function]}
          type="submit"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M18.3 5.71a.9959.9959 0 00-1.41 0L12 10.59 7.11 5.7a.9959.9959 0 00-1.41 0c-.39.39-.39 1.02 0 1.41L10.59 12 5.7 16.89c-.39.39-.39 1.02 0 1.41.39.39 1.02.39 1.41 0L12 13.41l4.89 4.89c.39.39 1.02.39 1.41 0 .39-.39.39-1.02 0-1.41L13.41 12l4.89-4.89c.38-.38.38-1.02 0-1.4z"
            />
          </svg>
        </button>
      </div>
    </div>
    <div
      className="sc-iGgWBj sc-ikkxIA fgQoGq kFUUjo"
    >
      Ingresa el teléfono vinculado a tu cuenta. Enviaremos un WhatsApp/SMS con el enlace para restablecer tu contraseña.
    </div>
    <div
      className="sc-esYiGF fWtARg"
      id="campoTelefono"
    >
      <div
        className="sc-kpDqfm sc-fXSgeo bTCkFB jMgeTG"
        id="campoTelefono-etiqueta"
      >
        Teléfono
      </div>
      <div
        className="sc-kpDqfm sc-dCFHLb bTCkFB ihOnyP"
        id="campoTelefono-contenedorCampoDeTexto"
      >
        <input
          autoFocus={true}
          className="sc-fhzFiK hJPwmC"
          data-indice={null}
          disabled={false}
          id="campoTelefono-campoDeTexto"
          maxLength={15}
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onInput={[Function]}
          onKeyPress={[Function]}
          placeholder=""
          readOnly={false}
          rows={null}
          type="number"
          value=""
        />
      </div>
    </div>
    <div
      className="sc-dAbbOL fFiTcC"
    >
      <div>
        Soy un contenedor simulado
      </div>
    </div>
  </div>
</div>
`;
